src/Controller/Component/FlashComponent.php

```diff
@@ -150,11 +150,14 @@ public function set($message, array $options = [])
      *
      * @param string $name Element name to use.
      * @param array $args Parameters to pass when calling `FlashComponent::set()`.
-     * @return void
+     * @return \Cake\Http\Session|void
```

Comment:
Then you need to use `|null` IMO
void cannot coexist with anything else as it disallows return value use/checking completely.
Which clearly is not the case when you start allowing this.